base/strings/basic.jl

```diff
@@ -269,11 +285,32 @@ function prevind(s::AbstractString, i::Integer)
     return 0 # out of range
 end
 
+function prevind(s::AbstractString, i::Integer, nchar::Integer)
+    nchar > 0 || throw(ArgumentError("nchar must be greater than 0"))
+    e = endof(s)
+    j = Int(i)
+    j < 1 && return 0
+    while nchar > 0
+        if j > e
+            j = e
+        else
+            j -= 1
+            while j >= 1 && !isvalid(s,j)
+                j -= 1
+            end
+        end
+        j < 1 && return j
```

Comment:
Doesn't the one-argument function return `0` in that case?

BTW, the CI failures are due to whitespace in tests.

Reply:
It gets triggered only if `nchar>1` so it did not apply (for `nchar=1` everything was OK) and previously I have made it consistent with `String` implementation. But you are right that it is more consistent to return `0`. Fixed

Reply:
So that means this line isn't tested? It should definitely be, especially since that code is quite tricky (are there other cases like this?).

Reply:
OK. I have changed the implementation to satisfy a contract that iterating `nextind`/`prevind` `k` times should give the same as `nextind`/`prevind` with `nchar=k`. It is also fully tested.